src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -73,71 +72,34 @@ public PluginRemoting(String hpiRemoteUrl){
     public PluginRemoting(File pomFile){
         this.pomFile = pomFile;
     }
-	
-    private String retrievePomContent() throws PluginSourcesUnavailableException{
-        if (hpiRemoteUrl != null) {
-            return retrievePomContentFromHpi();
-        } else {
-            return retrievePomContentFromXmlFile();
-        }
-    }
 
-    private String retrievePomContentFromHpi() throws PluginSourcesUnavailableException {
+    private String retrievePomContent() {
         try {
-	    return retrievePomContentFromHpiRemoteUrl(new URL(hpiRemoteUrl));
-	} catch (MalformedURLException e) {
-	    LOGGER.log(Level.WARNING, "HPI reference {0} is not a remote URL", hpiRemoteUrl);
-	    return retrievePomContentFromHpiFileReference();
-	}
-    }
-
-    private String retrievePomContentFromHpiRemoteUrl(URL url) throws PluginSourcesUnavailableException {
-    	try {
-    	    return retrievePomContentFromInputStream(url.openStream());
-	} catch (IOException e) {
-	    LOGGER.log(Level.WARNING, "Failed to retrieve POM content from HPI remote URL", e);
-          throw new PluginSourcesUnavailableException("Failed to retrieve POM content from HPI remote URL", e);
-	}
-    }
-
-    private String retrievePomContentFromHpiFileReference() throws PluginSourcesUnavailableException {
-    	try {
-	    String fileReference = hpiRemoteUrl.replaceAll("jar:", "").replaceAll("!/name.hpi", "");
-    	    return retrievePomContentFromInputStream(FileUtils.openInputStream(new File(fileReference)));
-	} catch (IOException e) {
-	    LOGGER.log(Level.WARNING, "Failed to retrieve POM content from HPI file reference", e);
-          throw new PluginSourcesUnavailableException("Failed to retrieve POM content from HPI file reference", e);
-	}
+            if (hpiRemoteUrl != null) {
+                return retrievePomContentFromHpi();
+            } else {
+                return retrievePomContentFromXmlFile();
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
-    private String retrievePomContentFromInputStream(InputStream pluginUrlStream) throws PluginSourcesUnavailableException {
-    	try (ZipInputStream zin = new ZipInputStream(pluginUrlStream)) {
-            ZipEntry zipEntry = zin.getNextEntry();
-            while(!zipEntry.getName().startsWith("META-INF/maven") || !zipEntry.getName().endsWith("pom.xml")){
-                zin.closeEntry();
-                zipEntry = zin.getNextEntry();
+    private String retrievePomContentFromHpi() throws IOException {
+        URL url = new URL(hpiRemoteUrl);
+        try (InputStream is = url.openStream(); ZipInputStream zis = new ZipInputStream(is)) {
+            ZipEntry ze;
+            while ((ze = zis.getNextEntry()) != null) {
+                if (ze.getName().startsWith("META-INF/maven") && ze.getName().endsWith("pom.xml")) {
```

Comment:
should make no difference but worth a hardening here (in case the archive somehow starts bundling a `dependecny-reduced-pom.xml` or the like.)?

```suggestion
                if (ze.getName().startsWith("META-INF/maven/") && ze.getName().endsWith("/pom.xml")) {
```